cmd/test-utils/common.go

```diff
@@ -0,0 +1,107 @@
+package cmdtest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+)
+
+var srcdBin = fmt.Sprintf("../../../build-integration/engine_%s_%s/srcd", runtime.GOOS, runtime.GOARCH)
+
+type IntegrationSuite struct {
+	suite.Suite
+}
+
+func init() {
+	if os.Getenv("SRCD_BIN") != "" {
+		srcdBin = os.Getenv("SRCD_BIN")
+	}
+}
+
+func (s *IntegrationSuite) runCommand(ctx context.Context, cmd string, args ...string) (*bytes.Buffer, error) {
+	args = append([]string{cmd}, args...)
+
+	var out bytes.Buffer
+
+	command := exec.CommandContext(ctx, srcdBin, args...)
+	command.Stdout = &out
+	command.Stderr = &out
+
+	return &out, command.Run()
+}
+
+var logMsgRegex = regexp.MustCompile(`.*msg="(.+)"`)
+
+func (s *IntegrationSuite) ParseLogMessages(memLog *bytes.Buffer) []string {
```

Comment:
is it possible to use TraceLogMessages here? Code looks similar

Reply:
Initially, that was what I wanted to do. The only thing is that `TraceLogMessages` works because it changes the format to JSON just before running `fn`. This is not possible for integration testing, unless we provide a `--log-format` flag to run `srcd` commands. That's the reason I opened #244.